lte/gateway/c/core/oai/test/amf/test_amf.cpp

```diff
@@ -95,17 +95,24 @@ uint8_t NAS5GPktSnapShot::registration_reject[4] = {0x00, 0x00, 0x00, 0x00};
 
 uint8_t NAS5GPktSnapShot::security_mode_reject[4] = {0x7e, 0x00, 0x5f, 0x24};
 
-TEST(test_amf_nas5g_pkt_process, test_amf_ue_register_req_msg) {
-  NAS5GPktSnapShot nas5g_pkt_snap;
+class AmfNas5GTest : public ::testing::Test {
+ protected:
+  NAS5GPktSnapShot* nas5g_pkt_snap;
   RegistrationRequestMsg reg_request;
-  bool decode_res = false;
-
-  uint32_t len = nas5g_pkt_snap.get_reg_req_buffer_len();
+  bool decode_res;
+  virtual void SetUp() {
+    nas5g_pkt_snap = new NAS5GPktSnapShot();
+    decode_res     = false;
+    memset(&reg_request, 0, sizeof(RegistrationRequestMsg));
```

Comment:
initialize using `reg_request = {0};`

Reply:
initializing with RegistrationRequestMsg reg_request = {0};  will give the following error.
![error](https://user-images.githubusercontent.com/88074557/135207032-de41990a-5909-4bf9-a120-20cd9703ac42.png)

so, initialized as RegistrationRequestMsg reg_request = {}
![tests](https://user-images.githubusercontent.com/88074557/135207211-ba245b89-c611-4ed5-80c1-1db5ceff3b48.png)

Reply:
@Krishnavamsi-wavelabs why not initialize with `new` constructor? the same way you're initializing `NAS5GPktSnapShot`

Reply:
@ardzoht  ,
addressed review comment.
both NAS5GPktSnapShot,RegistrationRequestMsg structures defined normal member variables, dynamic memory allocation not needed.

Reply:
![Screenshot (51)](https://user-images.githubusercontent.com/88074557/135485983-b91de99a-5307-4904-808c-6775865a76dd.png)